### PR TITLE
Sympy floor ceil

### PR DESCRIPTION
**Pull request by TizianoDeMatteis**
When a symbolic expression is converted to C++, the  `sympy_intdiv_fix` to simplify ceiling/floor expression.

This happens, for example, when the code for a map is generated. If map ranges are defined in terms of ceiling (for example), the code generated is ok as long as the range expression is not complicated (I don't how to define complicate it). For example:
- a map range defined as `"n": "0:ceiling(ONNX_num_filters/6) - 1"` is then codegenerated as `for (int n0 = 0; n0 < int_ceil(ONNX_num_filters, 6); n0 += 1)`, where the ceiling function as been replaced with `int_ceil` (FPGA backend)
- something like the following `"n": "0:ceiling((ONNX_in_features_x - 4)*(ONNX_in_features_y - 4)/576) - 1"` is instead left as it is, codegenerating also the `ceiling` function call, hence compiler error

This PR, matches also generic expressions at the numerator

Note: This also Solves #570.
This solution takes into account that the OCL Keyword Remover updates the AST node, by maintaining the same type: in this particular case, it is an ast.Name, where the id is something like *smtg.

Therefore, in the type inference, if we find a leading * we simply remove it before inferring the type.

### CODE REVIEW

File: dace/symbolic.py [hunk 1]
```diff
@@ -516,6 +516,7 @@ def sympy_intdiv_fix(expr):
     b = sympy.Wild('b', properties=[lambda k: k.is_Symbol or k.is_Integer])
     c = sympy.Wild('c')
     d = sympy.Wild('d')
+    e = sympy.Wild('e', properties=[lambda k: isinstance(k, sympy.Basic)])
```
Comment (tbennun) [inline dace/symbolic.py:519]: Even integers are sympy.Basic (`isinstance(sp.sympify('1'), sp.Basic)` --> `True`). Would this always work?

Comment (alexnick83) [reply dace/symbolic.py:519]: Has this been resolved? We can probably add a test. If it is an issue, then I suggest to either do `isinstance(k, sympy.Expr)` or perhaps even better `isinstance(k, sympy.Basic) and not isinstance(k, sympy.Atom)`. 
From the SymPy documentation about the `Atom` class:
> A parent class for atomic things. An atom is an expression with no subexpressions.
>
> Examples
> 
> Symbol, Number, Rational, Integer, … But not: Add, Mul, Pow, …

**Review by alexnick83 — CHANGES_REQUESTED** [review]
Looks ok to me apart from verifying whether `sympy.Basic` is an issue, as Tal suggested.

File: dace/symbolic.py (hunk 1 above)
Comment (TizianoDeMatteis) [reply dace/symbolic.py]: It should be solved now, I've explicitly added to not match Atom expressions

**Review by tbennun — APPROVED** [review]
Approved if tests pass

**Pull request merged**